### ISSUE TITLE
Option to "record"/visualize clicks

### ISSUE THREAD

**Issue by atomicptr** [phw/peek#159]
I'm usually using peek to show people something in a program, but it's not always clear when I pressed a mouse button. An option to visualize a click in some way would be really helpful.

**Comment by phw** [comment]
Definitely useful, but not entirely sure I will implement this. You can easily get this by using a tool like key-mon, see also https://github.com/phw/peek#how-can-i-capture-mouse-clicks-andor-key-strokes

I keep this feature request open and look at my options for doing this directly in Peek.

**Comment by peteruithoven** [comment]
I also like the visual style of [screenkey](https://github.com/wavexx/screenkey), but is has worked very unstable for me.

**Comment by phw** [comment]
@peteruithoven Yes, that's another option. But I don't think screenkey supports showing clicks, only keypresses.

**Comment by peteruithoven** [comment]
Good point: https://github.com/wavexx/screenkey/issues/43, https://github.com/wavexx/screenkey/issues/18

**Comment by khalilfazal** [comment]
Hey guys, I created this PR here: https://github.com/wavexx/screenkey/pull/45
Hopefully this goes a long way towards resolving this issue too 😄

**Comment by luzpaz** [comment]
@khalilfazal do you plan to submit a PR to peek as well ?

**Comment by khalilfazal** [comment]
I haven't worked on this project in a while. There is just the one PR. The first post there is up-to-date with what's in the PR.

**Comment by rugk** [comment]
In my case the overlay/mouse clicks by key-mon (as recommended here) were not recorded (in addition to the mouse cursor that is missing: https://github.com/phw/peek/issues/430). Possibly, because of Wayland or so.

**Comment by aoloe** [comment]
this might be easy to integrate into peek...

https://github.com/arp242/find-cursor

all i'm missing is how to capture the mouse click...

**Comment by rugk** [comment]
From the FAQ:

> But you could install an external tool like key-mon which is usually included in most distributions,

Well… at least in the currently stable Fedora version it is not. [Actually I see](https://repology.org/project/key-mon/versions) it [has no maintainer and has been orphaned](https://src.fedoraproject.org/rpms/key-mon/c/3d91f462ed2abbe06d6e60dcc13b7fedb5de7398?branch=rawhide) there. Also [it is not really included in many distros](https://pkgs.org/search/?q=key-mon), mostly Debian and Ubuntu.
So this is not really a convincing reasoning IMHO.

Independently to this, of course it's just a lot more convenient to have this feature built-in than to start a third-party (CLI?) program. And you may enable/disable it in the UI.

**Comment by jurgen-publiq** [comment]
I agree here that we should adhere to the linux philosophy: let's have a lot of small tools that do one thing great.
It would be better to have someone [ maintain | fork | create something like ] key-mon than to have this integrated into peek. There may be other conditions where a key-higlighter kan be useful (e.g. classroom tutorials, live streams, etc...). Also, I don't think there are much resources to be won if you embed it into peek. It will make peek more sluggish for most.
